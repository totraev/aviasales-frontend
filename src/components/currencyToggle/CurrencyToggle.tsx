import React, { SFC } from 'react';
import styles from './CurrencyToggle.module.css';

const CurrencyToggle: SFC = () => (
  <div className={styles.toggle}>
    <button className={styles.btnActive}>RUB</button>
    <button className={styles.btnDefault}>USD</button>
    <button className={styles.btnDefault}>EUR</button>
  </div>
);

export default CurrencyToggle;
