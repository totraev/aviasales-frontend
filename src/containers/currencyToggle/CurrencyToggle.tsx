import React, { SFC } from 'react';
import styles from './CurrencyToggle.module.css';

import classNames from 'classnames';
import { withHandlers } from 'recompose';

import { ICurrencyToggleProps, IHandlers, IInnerProps } from './CurrencyToggle.types';

export const CurrencyToggle: SFC<IInnerProps> = ({ handleBtnClassName, handleClick }) => (
  <div className={styles.toggle}>
    <button
      className={handleBtnClassName('rub')}
      onClick={handleClick('rub')}
      children="RUB"
    />
    <button
      className={handleBtnClassName('usd')}
      onClick={handleClick('usd')}
      children="USD"
    />
    <button
      className={handleBtnClassName('eur')}
      onClick={handleClick('eur')}
      children="EUR"
    />
  </div>
);

export default withHandlers<ICurrencyToggleProps, IHandlers>({
  handleBtnClassName: ({ currency: activeCurr }) => (curr) => classNames({
    [styles.btnActive]: activeCurr === curr,
    [styles.btnDefault]: activeCurr !== curr,
  }),
  handleClick: ({ onChange }) => (currency) => () => onChange(currency),
})(CurrencyToggle);
