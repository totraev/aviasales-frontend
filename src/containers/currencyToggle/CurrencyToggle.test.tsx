import React from 'react';
import renderer from 'react-test-renderer';

import { CurrencyToggle } from './CurrencyToggle';

describe('CurrencyToggle component', () => {
  it('Should renders correctly', () => {
    const mockFunc: any = () => null;
    const tree = renderer.create(
      <CurrencyToggle
        currency="rub"
        onChange={null}
        handleBtnClassName={mockFunc}
        handleClick={mockFunc}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
