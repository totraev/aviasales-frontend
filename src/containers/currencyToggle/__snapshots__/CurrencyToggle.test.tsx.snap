// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyToggle component Should renders correctly 1`] = `
<div
  className="toggle"
>
  <button
    className={null}
    onClick={null}
  >
    RUB
  </button>
  <button
    className={null}
    onClick={null}
  >
    USD
  </button>
  <button
    className={null}
    onClick={null}
  >
    EUR
  </button>
</div>
`;
